airflow/datasets/manager.py

```diff
@@ -66,6 +67,7 @@ def register_dataset_change(
             )
         )
         session.flush()
+        Stats.incr("dataset.changed_datasets")
```

Comment:
```suggestion
        Stats.incr("dataset.updates")
```

I like this change! Metrics feel a bit like an afterthought in many changes. I do however suggest a different name, because:

- The docs speak about "updates" instead of changes: https://airflow.apache.org/docs/apache-airflow/stable/concepts/datasets.html, would stick to that terminology for consistency
- Remove "datasets" because it's already in the name

Reply:
I agree